internal/cmd/agent/root.go

```diff
@@ -4,53 +4,61 @@ import (
 	"fmt"
 	"log"
 	"net/http"
-	"time"
 
 	"github.com/spf13/cobra"
 
+	ctrl "sigs.k8s.io/controller-runtime"
+	clog "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
 	command "github.com/rancher/fleet/internal/cmd"
 	"github.com/rancher/fleet/pkg/version"
 )
 
+type UpstreamOptions struct {
+	Kubeconfig string `usage:"kubeconfig file for agent's cluster"`
+	Namespace  string `usage:"system namespace is the namespace, the agent runs in, e.g. cattle-fleet-system" env:"NAMESPACE"`
+}
+
 type FleetAgent struct {
 	command.DebugConfig
-	Kubeconfig      string `usage:"kubeconfig file"`
-	Namespace       string `usage:"namespace to watch" env:"NAMESPACE"`
-	AgentScope      string `usage:"An identifier used to scope the agent bundleID names, typically the same as namespace" env:"AGENT_SCOPE"`
-	CheckinInterval string `usage:"How often to post cluster status" env:"CHECKIN_INTERVAL"`
+	UpstreamOptions
+	AgentScope string `usage:"An identifier used to scope the agent bundleID names, typically the same as namespace" env:"AGENT_SCOPE"`
 }
 
+var setupLog = ctrl.Log.WithName("setup")
+
 func (a *FleetAgent) Run(cmd *cobra.Command, args []string) error {
+	zopts := zap.Options{
+		Development: true,
+	}
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&zopts)))
+	ctx := clog.IntoContext(cmd.Context(), ctrl.Log)
```

Comment:
dumb question: should this use `setupLog`?

Reply:
No, this is the logger in the context. I want one there in every process, in case share code uses it in some package.